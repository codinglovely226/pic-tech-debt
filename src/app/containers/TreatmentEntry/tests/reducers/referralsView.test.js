/* eslint-disable */

import ReferralsViewReducer from '../../reducers/referralsView';

import {
  resetReferralsView,
  updateReferralsViewReferee,
  updateReferralsViewRefereeSite,
  updateReferralsViewSpecialty,
} from '../../actions/case';

describe('ReferralsViewReducer', () => {
  it('returns the initial state', () => {
    expect(ReferralsViewReducer(undefined, {})).toMatchSnapshot();
  });
  it('reset referrals view', () => {
    expect(ReferralsViewReducer({'referee': 'Test', 'specialty': 'Test', 'refereeSite': 1}, resetReferralsView())).toMatchSnapshot();
  });
  it('update referrals view referee', () => {
    expect(ReferralsViewReducer({'referee': 'Test', 'specialty': 'Test', 'refereeSite': 1}, updateReferralsViewReferee('Test Referee'))).toMatchSnapshot();
  });
  it('update referrals view referee site', () => {
    expect(ReferralsViewReducer({'referee': 'Test', 'specialty': 'Test', 'refereeSite': 1}, updateReferralsViewRefereeSite('Test Referee Site'))).toMatchSnapshot();
  });
  it('update referrals view specialty', () => {
    expect(ReferralsViewReducer({'referee': 'Test', 'specialty': 'Test', 'refereeSite': 1}, updateReferralsViewSpecialty('Test Specialty'))).toMatchSnapshot();
  });
});
