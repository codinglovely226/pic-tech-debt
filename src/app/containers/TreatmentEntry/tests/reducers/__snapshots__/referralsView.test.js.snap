exports[`ReferralsViewReducer reset referrals view 1`] = `
Object {
  "referee": "",
  "refereeSite": "",
  "specialty": "",
}
`;

exports[`ReferralsViewReducer returns the initial state 1`] = `
Object {
  "referee": "",
  "refereeSite": "",
  "specialty": "",
}
`;

exports[`ReferralsViewReducer update referrals view referee 1`] = `
Object {
  "referee": "Test Referee",
  "refereeSite": 1,
  "specialty": "Test",
}
`;

exports[`ReferralsViewReducer update referrals view referee site 1`] = `
Object {
  "referee": "Test",
  "refereeSite": "Test Referee Site",
  "specialty": "Test",
}
`;

exports[`ReferralsViewReducer update referrals view specialty 1`] = `
Object {
  "referee": "Test",
  "refereeSite": 1,
  "specialty": "Test Specialty",
}
`;
